src/Momento.Sdk/Responses/CacheGetBatchResponse.cs

```diff
@@ -1,29 +1,84 @@
 ﻿using System.Collections.Generic;
-using System.Linq;
+using Momento.Sdk.Exceptions;
 
 namespace Momento.Sdk.Responses;
 
-public class CacheGetBatchResponse
+public abstract class CacheGetBatchResponse
 {
-    public List<CacheGetResponse> Responses { get; }
-
-    public CacheGetBatchResponse(IEnumerable<CacheGetResponse> responses)
+    public class Success : CacheGetBatchResponse
     {
-        this.Responses = new(responses);
-    }
+        public List<CacheGetResponse> Responses { get; }
 
-    public IEnumerable<CacheGetStatus> Status
-    {
-        get => Responses.Select(response => response.Status);
-    }
+        public Success(IEnumerable<CacheGetResponse> responses)
+        {
+            this.Responses = new(responses);
+        }
 
-    public IEnumerable<string?> Strings()
-    {
-        return Responses.Select(response => response.String());
+        public IEnumerable<CacheGetStatus> Status
+        {
+            get
+            {
+                var ret = new List<CacheGetStatus>();
+                foreach (CacheGetResponse response in Responses) {
+                    if (response is CacheGetResponse.Hit hitResponse) {
```

Comment:
These look better/safer now, so thanks for changing them.

I'm still feeling iffy about this part of the API as a whole, so I think we should revisit it and have a convo about it later this week, but I think it's okay for now for this PR.

Reply:
Agreed, the batch stuff needs some love and we should take a closer look when time allows.